irohad/torii/impl/command_service_transport_grpc.cpp

```diff
@@ -0,0 +1,206 @@
+     * @return message
+     */
+    std::string formErrorMessage(
+        const std::vector<shared_model::crypto::Hash> &tx_hashes,
+        const std::string &error) {
+      if (tx_hashes.size() == 1) {
+        return (boost::format("Stateless invalid tx, error: %s, hash: %s")
+                % error % tx_hashes[0].hex())
+            .str();
+      }
+
+      std::string folded_hashes =
+          std::accumulate(tx_hashes.begin(),
+                          tx_hashes.end(),
+                          std::string(),
+                          [](auto &&acc, const auto &h) -> std::string {
+                            return acc + h.hex() + ", ";
+                          });
+
+      // remove leading ", "
+      folded_hashes.resize(folded_hashes.size() - 2);
+
+      return (boost::format(
+                  "Stateless invalid tx in transaction sequence, error: %s\n"
+                  "Hash list: [%s]")
+              % error % folded_hashes)
+          .str();
+    }
+  }  // namespace
+
+  grpc::Status CommandServiceTransportGrpc::ListTorii(
+      grpc::ServerContext *context,
+      const iroha::protocol::TxList *request,
+      google::protobuf::Empty *response) {
+    auto tx_list_builder = shared_model::proto::TransportBuilder<
+        shared_model::interface::TransactionSequence,
+        shared_model::validation::DefaultUnsignedTransactionsValidator>();
+
+    tx_list_builder.build(*request).match(
+        [this](
+            iroha::expected::Value<shared_model::interface::TransactionSequence>
+                &tx_sequence) {
+          this->command_service_->handleTransactionList(tx_sequence.value);
+        },
+        [this, request](auto &error) {
+          auto &txs = request->transactions();
+          if (txs.empty()) {
+            log_->warn("Received no transactions. Skipping");
+            return;
+          }
+          using HashType = shared_model::crypto::Hash;
+
+          std::vector<HashType> hashes;
+          std::transform(
+              txs.begin(),
+              txs.end(),
+              std::back_inserter(hashes),
+              [](const auto &tx) {
+                return shared_model::crypto::DefaultHashProvider::makeHash(
+                    shared_model::proto::makeBlob(tx.payload()));
+              });
+
+          auto error_msg = formErrorMessage(hashes, error.error);
+          // set error response for each transaction in a sequence
+          std::for_each(
+              hashes.begin(), hashes.end(), [this, &error_msg](auto &hash) {
+                status_bus_->publish(
+                    status_factory_->makeStatelessFail(hash, error_msg));
+              });
+        });
+    return grpc::Status::OK;
+  }
+
+  grpc::Status CommandServiceTransportGrpc::Status(
+      grpc::ServerContext *context,
+      const iroha::protocol::TxStatusRequest *request,
+      iroha::protocol::ToriiResponse *response) {
+    auto status = command_service_->getStatus(
```

Comment:
Is it possible to use `getTransport`? Like `*response = command_service_->getStatus(shared_model::crypto::Hash(request->tx_hash()))->getTransport();`